scripts/pre-push.sh

```
@@ -0,0 +1,50 @@
+
+echo "Running lints..."
+make lint
+LINT_RESULT=$?
+
+if [ $LINT_RESULT -ne 0 ]; then
+    echo "Linting failed. Push aborted."
+    exit 1
+fi
+
+# Check for WIP commits (optional, from the example)
+remote="$1"
```

Shellcheck (0.10.0)

[warning] remote appears unused. Verify use (or export if used externally).

(SC2034)